lib/src/style.dart

```diff
@@ -72,9 +77,11 @@ class ToggleStyle {
           ? this
           : ToggleStyle._(
               indicatorColor: other.indicatorColor ?? indicatorColor,
+              indicatorGradient: other.indicatorGradient ??
+                  (other.indicatorColor != null ? null : indicatorGradient),
               backgroundColor: other.backgroundColor ?? backgroundColor,
               backgroundGradient: other.backgroundGradient ??
-                  (other.backgroundColor == null ? null : backgroundGradient),
+                  (other.backgroundColor != null ? null : backgroundGradient),
```

Comment:
@denisgl7 This line can cause a breaking change. That's why I need to take another look at it.